document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -98,16 +98,21 @@ public class PostgresCollection implements Collection {
       CloseableIterator.emptyIterator();
 
   private final PostgresClient client;
-  private final String collectionName;
+  private final PostgresTableIdentifier tableIdentifier;
   private final PostgresSubDocumentUpdater subDocUpdater;
   private final PostgresQueryExecutor queryExecutor;
   private final UpdateValidator updateValidator;
 
   public PostgresCollection(final PostgresClient client, final String collectionName) {
+    this(client, PostgresTableIdentifier.parse(collectionName));
+  }
+
+  PostgresCollection(final PostgresClient client, final PostgresTableIdentifier tableIdentifier) {
     this.client = client;
-    this.collectionName = collectionName;
-    this.subDocUpdater = new PostgresSubDocumentUpdater(new PostgresQueryBuilder(collectionName));
-    this.queryExecutor = new PostgresQueryExecutor(collectionName);
+    this.tableIdentifier = tableIdentifier;
```

Comment:
As `collectionName` is tied to the `Collection` object, if we could consider resolving collectionName here, we would only need to resolve it once, instead of resolving it every time it is accessed via the toString method of tableIdentifier. This is a very minor point, but I wanted to share it.

Reply:
Fair, but I'd prefer to keep the object form unless you object. Thinking - the overhead of toString is close to zero (basically an optional unwrap in the typical case or a string concat in the schema case), especially relative to the remote DB call that's part of the same sequence.